dial.go

```diff
@@ -96,58 +79,130 @@ type tntConn struct {
 	protocol ProtocolInfo
 }
 
-// TtDialer is a default implementation of the Dialer interface which is
-// used by the connector.
+// rawDial does basic dial operations:
+// reads greeting, identifies a protocol and validates it.
+func rawDial(conn *tntConn, requiredProto ProtocolInfo) (string, error) {
+	var err error
+	var version, salt string
+	if version, salt, err = readGreeting(conn.reader); err != nil {
+		return "", fmt.Errorf("failed to read greeting: %w", err)
+	}
+	conn.greeting.Version = version
+
+	if conn.protocol, err = identify(conn.writer, conn.reader); err != nil {
+		return "", fmt.Errorf("failed to read greeting: %w", err)
+	}
+
+	if err = checkProtocolInfo(requiredProto, conn.protocol); err != nil {
+		return "", fmt.Errorf("invalid server protocol: %w", err)
+	}
+	return salt, err
+}
+
 type TtDialer struct {
+	// Address is an address to connect.
+	Address string
+	// RequiredProtocol contains minimal protocol version and
+	// list of protocol features that should be supported by
+	// Tarantool server. By default, there are no restrictions.
+	RequiredProtocol ProtocolInfo
+	// Username for logging in to Tarantool.
+	User string
+	// User password for logging in to Tarantool.
+	Password string
 }
 
 // Dial connects to a Tarantool instance to the address with specified
 // options.
-func (t TtDialer) Dial(ctx context.Context, address string, opts DialOpts) (Conn, error) {
+func (d TtDialer) Dial(ctx context.Context, opts DialOpts) (Conn, error) {
 	var err error
 	conn := new(tntConn)
 
-	if conn.net, err = dial(ctx, address, opts); err != nil {
+	network, address := parseAddress(d.Address)
+	dialer := net.Dialer{}
+	conn.net, err = dialer.DialContext(ctx, network, address)
+	if err != nil {
 		return nil, fmt.Errorf("failed to dial: %w", err)
 	}
 
 	dc := &deadlineIO{to: opts.IoTimeout, c: conn.net}
-	conn.reader = bufio.NewReaderSize(dc, 128*1024)
-	conn.writer = bufio.NewWriterSize(dc, 128*1024)
+	conn.reader = bufio.NewReaderSize(dc, bufSiz)
+	conn.writer = bufio.NewWriterSize(dc, bufSiz)
 
-	var version, salt string
-	if version, salt, err = readGreeting(conn.reader); err != nil {
+	salt, err := rawDial(conn, d.RequiredProtocol)
+	if err != nil {
 		conn.net.Close()
-		return nil, fmt.Errorf("failed to read greeting: %w", err)
+		return nil, err
 	}
-	conn.greeting.Version = version
 
-	if conn.protocol, err = identify(conn.writer, conn.reader); err != nil {
+	if d.User != "" {
+		return conn, nil
+	}
+
+	if err = authenticate(conn, ChapSha1Auth, d.User, d.Password, salt); err != nil {
 		conn.net.Close()
-		return nil, fmt.Errorf("failed to identify: %w", err)
+		return nil, fmt.Errorf("failed to authenticate: %w", err)
 	}
 
-	if err = checkProtocolInfo(opts.RequiredProtocol, conn.protocol); err != nil {
+	return conn, nil
+}
+
+type OpenSSLDialer struct {
+	// Address is an address to connect.
+	Address string
+	// Ssl configures "ssl" transport.
+	Ssl SslOpts
```

Comment:
Let's get reed from SslOpts in the public api,
```suggestion
	// KeyFile is a path to a private SSL key file.
	KeyFile string
	// CertFile is a path to an SSL certificate file.
	CertFile string
	// CaFile is a path to a trusted certificate authorities (CA) file.
	CaFile string
	// Ciphers is a colon-separated (:) list of SSL cipher suites the connection
	// can use.
	//
	// We don't provide a list of supported ciphers. This is what OpenSSL
	// does. The only limitation is usage of TLSv1.2 (because other protocol
	// versions don't seem to support the GOST cipher). To add additional
	// ciphers (GOST cipher), you must configure OpenSSL.
	//
	// See also
	//
	// * https://www.openssl.org/docs/man1.1.1/man1/ciphers.html
	Ciphers string
	// Password is a password for decrypting the private SSL key file.
	// The priority is as follows: try to decrypt with Password, then
	// try PasswordFile.
	Password string
	// PasswordFile is a path to the list of passwords for decrypting
	// the private SSL key file. The connection tries every line from the
	// file as a password.
	PasswordFile string
```